src/main/java/org/apache/commons/compress/archivers/dump/TapeInputStream.java

```diff
@@ -294,6 +294,9 @@ public void resetBlockSize(final int recsPerBlock, final boolean isCompressed) t
             throw new IOException("Block with " + recsPerBlock + " records found, must be at least 1");
         }
         blockSize = RECORD_SIZE * recsPerBlock;
+        if (blockSize < 1) {
+            throw new IOException("Block size " + blockSize + " cannot be negative");
```

Comment:
Either the test or the error message is incorrect. If the test is correct then the error message should say `"Block size cannot be less than or equal to 0: " + blockSize"`